src/PresentationalComponents/TagsToolbar/ManageTags.js

```diff
@@ -1,36 +1,113 @@
-import { Modal } from '@patternfly/react-core/dist/js/components/Modal/Modal';
+import React, { useEffect, useState } from 'react';
+
+import { DEBOUNCE_DELAY } from '../../AppConstants';
 import PropTypes from 'prop-types';
-import React from 'react';
-import TagsList from './TagsList';
+import { TagModal } from '@redhat-cloud-services/frontend-components/components/TagModal';
 import { connect } from 'react-redux';
+import debounce from '../../Utilities/Debounce';
 import { injectIntl } from 'react-intl';
 import messages from '../../Messages';
+import { setSelectedTags } from '../../AppActions';
 
-const ViewHostAcks = ({ handleModalToggle, intl, selectedTags, isModalOpen, tags }) => <Modal
-    width='50%'
-    title={intl.formatMessage(messages.tagsCount, { count: selectedTags.length })}
-    isOpen={isModalOpen}
-    onClose={() => { handleModalToggle(false); }}
-    isFooterLeftAligned>
-    <TagsList tags={tags} />
-</Modal>;
-
-ViewHostAcks.propTypes = {
-    isModalOpen: PropTypes.bool,
-    selectedTags: PropTypes.array,
-    handleModalToggle: PropTypes.func,
-    tags: PropTypes.array,
-    intl: PropTypes.any
-};
+const ManageTags = ({ intl, toggleModal, fetchTags, selectedTags, setSelectedTags, totalTags, isOpen }) => {
+    const [tags, setTags] = useState();
+    const [loaded, setLoaded] = useState(false);
+    const [selected, setSelected] = useState([]);
+    const [perPage, setPerPage] = useState(10);
+    const [page, setPage] = useState(1);
+    const [searchText, setSearchText] = useState();
+    const debouncedSearchText = debounce(searchText, DEBOUNCE_DELAY);
+
+    useEffect(() => {
+        (async () => {
+            setPage(1);
+            setTags(await fetchTags(perPage, 1, null, searchText));
+            setLoaded(true);
+        })();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [debouncedSearchText]);
 
-ViewHostAcks.defaultProps = {
-    isModalOpen: false,
-    handleModalToggle: () => undefined
+    useEffect(() => {
+        (async () => {
+            setTags(await fetchTags(perPage, page, null, debouncedSearchText));
+            setLoaded(true);
+        })();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [perPage, page]);
+
+    useEffect(() => {
+        setSelected(selectedTags.length ? selectedTags.map(id => ({ id, selected: true })) : []);
+    }, [selectedTags, setSelected]);
+
+    return <TagModal
+        systemName={intl.formatMessage(messages.tagsModalTitle)}
+        {...loaded && {
+            loaded,
+            pagination: {
+                perPage,
+                page,
+                count: totalTags
+            },
+            rows: tags.map(tag => ({
+                id: tag.id,
+                selected: selected.find(selection => selection.id === tag.id),
+                cells: [tag.key, tag.value, tag.namespace]
```

Comment:
is the system count data available here?  i know its not in the mocks.. but we do display it in the toolbar, kinda seems silly the table doesn't also display it 🤔